content/js1/sprints/3/prep/index.md

```diff
@@ -8,4 +8,634 @@ backlog= 'Module-JS1'
+
+```bash
+Assertion failed: current output: 23:00 am, target output: 11:00 pm
+```
+
+{{<note type="exercise" title="Exercise 2.1">}}
+
+✏️ Write another assertion that checks `formatAs12HourClock` returns the target output when passed another _time input in the afternoon._
+
+{{</note>}}
+
+### ↙️ ↘️ Making a choice
+
+Our function works for morning inputs like `"08:00"`.
+In this case, the function returns the target output of `"08:00 am"` as required.
+
+However, at the moment, `formatAs12HourClock("23:00")`'s current output is `"23:00 am"`.
+
+> 💡 We need to execute some different logic when the time is beyond midday
+
+We can interpret this behaviour as a question:
+
+```mermaid
+
+flowchart LR
+
+A{Is the time before midday?} -- true --> B[Add am to time string]
+A -- false --> C[???]
+```
+
+We need to work out what to do in the case when the time input is later than midday.
+
+### 🧭 Describing the strategy
+
+Let's describe the strategy for dealing with an input that is after midday.
+
+Earlier we observed that when the time goes beyond midday then **we can subtract 12 from the hours time to get the new hours for the 12 hour clock time**.
+
+_Before_ writing code, we can define our approach in steps using a flowchart:
+
+Starting with an input like `"23:00"`:
+
+```mermaid
+
+flowchart TD
+
+A[Step 1: Extract the hours from the time '23:00', to get '23'] --> B[Step 2: Convert '23' to a number, to get 23]
+B --> C{Step 3\nCheck: Are the hours greater than 12?}
+C -- true --> D[Step 4: subtract 12 from the hours, to get 11]
+D --> E[Step 5: add 'pm' to the rest of the time, to get '11:00 pm']
+E --> F[Step 6: return the new time]
+```
+
+This approach involves running some **conditionally**. In this case, we're only going to continue doing Steps 4-6 if the condition **hours are greater than 12** is `true`.
+
+### ❓ Conditionally executing code
+
+In programming, we can use an `if` statement to execute some code when a given condition is `true`.  
+In JavaScript, we can write an `if` statement as follows:
+
+```js {title="main.js"}
+if (condition) {
+  // do some code in here
+}
+```
+
+The `if` statement consists of:
+
+1. `if` keyword: this is the start of the `if` statement
+2. `(condition)`: some conditional expression that is wrapped in parentheses.
+3. `{}`: a block statement: any code we want to execute if the condition is true goes inside the curly braces here
+
+We can represent this with a diagram too:
+
+```mermaid
+
+flowchart LR
+    IC{condition}
+    IB[Execute code in body]
+    EXIT([End of if statement])
+    IC --> |true| IB
+    IC -.-> |false| EXIT
+    IB --> EXIT
+```
+
+{{<note type="exercise" title="Exercise 2.1">}}
+Write a program that defines a variable `eleniAge`.
+
+If Eleni is less than 18, the program should print out nothing to the console.
+
+If Eleni is 18 years old or older, the program should print out:
+
+```bash
+Eleni is old enough to drive
+```
```

Comment:
This exercise feels a lot more like the "writing very forced code that doesn't really make sense" style of exercise we had before this rewrite.

The instruction "Write a program that defines a variable `eleniAge`" (and then implicitly assume that the trainee has written an assignment with a meaningful value, and the implicit idea that they're going to maybe change that value to experiment) assumes too much contextual knowledge about why we're making this variable, what values are being assigned, and how it should be changed.

I think it'd make more sense to either frame this as "write a function", or as "Here's some code, add to it / fill in the gaps". I think framing it as "write a function" probably makes a lot of sense, given what we've been teaching and where things are placed, but if not, I'd probably provide this script:

```js
const eleniAge = 10;
```

then instruct the trainees to copy that and write their conditional code underneath it.

And then later on have an explicit instruction "Change the value of `eleniAge` to `20` and check your code output what was intended"

Reply:
I wonder this is also an opportunity to use `console.assert` to validate the code?

Reply:
I agree - I just copied this over from the old syllabus. I'm going to replace it with something more interesting.